extensions/interactions/LogicProof/static/js/student.js

```diff
@@ -994,12 +994,16 @@ var logicProofStudent = (function() {
   *          }
   */
   var buildProof = function(proofString, questionInstance) {
-    if (proofString.slice(-1) === '\n') {
-      proofString = proofString.slice(0, proofString.length - 1);
-    }
     var lineStrings = proofString.split('\n');
+    // Ignore blank lines at the end

```

Comment:
Discussed offline. The principle of auto-removing unnecessary whitespace is a good one, but in this specific case, it looks like people generally aren't making this error, and fixing it would entail significant changes in how the line number mappings work for error reporting. So, decided to leave it as-is for now.